doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -396,9 +396,23 @@ To support validate interface speed on CLI side, a new field **supported_speeds*
 	; field                 = value
     ...
 	supported_speeds        = STRING                         ; supported speed list
+  speed                   = STRING                         ; operational speed
 
 An example value of supported_speeds could be "10000,25000,40000,100000".
 
+Before this feature, port speed in APP DB indicates both the configured speed and the operational speed. It is OK without this feature because port operational speed must be configured speed or port operational status is down. However, this is not true with this feature. Consider following flow:
+
+1. Configure port speed to 100G
+2. Configure advertised speed to 50G and enable auto negotiation
+3. Port operational speed turns to 50G
+4. Configure any port attribute, e.g mtu, portsyncd would put all port attributes to APP DB
+5. `show interface status` displays port speed as 100G which is incorrect
+
+To overcome this issue, following changes are required:
+
+1. Put port operational speed to STATE DB PORT_TABLE
```

Comment:
Agree to put oper state to STATE_DB, but i think as long as PORT table exists in APP_DB, we should put oper state in the APP_DB as well for consistency with previous model

Reply:
@prsunny is it because you still suspect there could be places where we may be still getting the "operational" speed from APP_DB?

Reply:
Thats right.

Reply:
Hi @prsunny and @prgeor , the problem is that we cannot guarantee that the oper speed value in APP_DB is always consistent with the actual oper speed in PHY. Please see issue https://github.com/Azure/sonic-swss/pull/1769. And I paste the issue description here:

The field "speed" in PORT_TABLE of APPL_DB has two meanings:
1. The configured port speed
2. The actual port speed

This is all right when SONiC has no auto negotiation feature because the actual speed must equal to the configured speed or the operational status is down. However, with port auto negotiation feature, if autoneg mode is enabled, the actual speed could be a value of adv_speeds which might not equal to "speed" in CONFIG DB. Now port auto negotiation feature handle it this way: 
1. if autoneg mode  is disabled, everything goes to the old way, the actual speed is always equal to field "speed"; 
2. if autoneg mode is enabled, and if operational status is up, get actual speed from SAI and update the field "speed" in APPL_DB.

For case#2, there is an issue. Think about following flow:
1. Set Ethernet0 speed=400000
2. Set Ethernet0 autoneg=on, adv_speeds=100000, and Ethernet0 is operational up. Now the speed of Ethernet0 in "show interfaces status" is 100G because orchagent get the actual speed from SAI. This is right.
3. Issue command: "config interface type Ethernet0 CR". (or configure any other port attribute)

Step#3 won't affect the auto negotiation status because the autoneg mode is enabled, force interface type will not be applied to SAI. But it triggers PORT_TABLE change and portsyncd will push all Ethernet0 related configuration to APPL_DB. So, speed of Ethernet0 in "show interface status" changed to 400G. And 400G is not the actual speed in PHY.

Reply:
i had offline discussion with @prsunny, we agree to keep configured speed in ConfigDB/AppDB and operational speed in StateDB